packages/core/src/utils.ts

```diff
@@ -92,7 +93,13 @@ export async function copyWalletAddress(text: string): Promise<void> {
   }
 }
 
-export const chainIdToHex = (chain: Chain): Chain => typeof chain.id === 'number' ? { ...chain, id: `0x${chain.id.toString(16)}` } : chain
+export function chainIdToHex(chains : Chain[] | ChainWithDecimalId[] ): 
+Chain[] {
+  return chains.map(({ id, ...rest }) => { 
+    id = typeof id === 'number' ?  `0x${Math.abs(id).toString(16)}` : id
```

Comment:
Why the need for `Math.abs`? I think if we want to ensure that it is a positive number, we should add that to the validation with the `positive` function.